### ISSUE TITLE
H2O is incorperated in Allison model via Lesne in all her spreadsheet and all figures 

### ISSUE THREAD

**Issue by PennyWieser** [kaylai/VESIcal#62]
TLDR - Allisons supplementary spreadsheet incorporates H2O (I think in the Lesne model) - as the python script is just pure CO2, you get drastically different answers. I'm not sure its sensible to have an option for just CO2 as you get such different results, and her system she calibrated on was wet (2-5wt% H2O) - so either we need to make it extremely clear users must couple it with their own H2O model (else Psat can be wrong be a factor of 4), or add the Lesne model she uses.  

i've been using the spreadsheet Allison provides in her supplement to calculate PSat to test the python code, and the answer in her spreadsheet depends on the H2O content you enter. In contrast, when i load the same data into VESIcal it returns the same pressure regardless of the H2O (592 bar, As CO2-only). This never matches the pressure that allison's spreadsheet returns, even for H2O=0 wt% in Allison, Psat=799, in VESIcal=592 bar. Here is a screenshot, where the inputted pressure is what allisons spreadsheet gave out vs. that calculated by VESIcal
![image](https://user-images.githubusercontent.com/65668673/83290501-99996500-a1de-11ea-80ce-c188ff71fcc7.png)

I've re-read the allison paper in more detail. Although her model doesn't use H2O, I think she always intended it to be used alongside the Lesne model for water, never in isolation, as all her melts are really quite wet (2.1-5.3 wt% H2O). For example, for the isobars she calculates in Fig. 9, she uses the Lesne model for water
"One set of isobars is calculated using MagmaSat, while the other uses power law
fits for each volatile phase: Eq. 2 from this study for CO2 and the equation from Lesne et al. (2011a) for water."

Similarly, when talking about calculation Pentrap she says:

"We recalculate CO2 saturation from the thermodynamic model, using the parameters
listed in Table 4 for Stromboli, and use the power law ft
for Stromboli from Lesne et al. (2011a) for H2O solubility"

I don't think it really makes sense to let users use Allison without insisting they use a H2O model as well - especially given that the VESIcal answer doesn't even match her spreadsheet if you use H2O=0. For the compositions the model is actually calibrated at, the Psats calculated by VESIcal can be up to a factor of 4 wrong which is pretty dramatic. 

Let me know if i've misinterpreted something massively here, and all is actually fine though! I've attached her spreadsheet and the test compositions I loaded which are sunset crater compositions
[Test_Allison.xlsx](https://github.com/kaylai/VESIcal/files/4703560/Test_Allison.xlsx)

[Allison_Spreadsheet.xlsx](https://github.com/kaylai/VESIcal/files/4703522/Allison_Spreadsheet.xlsx)

**Comment by simonwmatthews** [comment]
I'll reply more fully when I have time. But in brief, you can define a MixedFluid model using Allison for carbon and any model you choose for H2O.

**Comment by PennyWieser** [comment]
Great, that makes sense. When you have a chance it would be great to know how to implement MixedFluid into this "satPs = myfile.calculate_saturation_pressure(temperature=1200, model='AllisonCarbon', model_loc='sunset'" so that I can continue testing the models vs. the provided spreadsheet

**Comment by simonwmatthews** [comment]
See docs for MixedFluid object for more info, but using
model = MixedFluid({'CO2':AllisonCarbon,'H2O':name_of_water_model}) 
will generate a custom mixed fluid model (as on lines 5875-5886 of current version of code).
You then pass this model object instead of a string for the model variable in the call to calculate_saturation_pressure().

You can access the docs by looking at the module code directly, or by adding "?" or "??" after the method name when calling it in jupyter lab.

The Lesne model doesn't exist in the code as I originally coded up the Allison model for a very specific purpose. It probably wouldn't take much effort to add the Lesne model, you could use the Allison model as a template. However, I expect you will be able to recreate the behaviour using any other water model.

As for letting people use the model without Lesne- I think we should allow people to use the code however they want, I think the burden is on them to make sure what they're doing is sensible and appropriate. However, it would be sensible to flag that up with a warning. I'll think of how best to implement this in the current calibration_check structure.

**Comment by kaylai** [comment]
Hi gang. Nice discussion here. I agree we should allow users to use the model how they choose (as per Simon's response), but I also agree with Penny that using a pure AllisonCarbon model seems not to be the way the model was intended to be used. Flagging this up in a warning I think is imperative. 

I can look into adding the Lesne (2011) H2O model to VESIcal, but they present three different equations all for different specific volcanoes (Vesuvius, Etna, and Stromboli). Do we know which was used in Allison et al?

If we need more info from Chelsea Allison, she is a friend of mine, so we could simply ask.

**Comment by PennyWieser** [comment]
Hi Kayla, I had a good look and can't find which one, unless she has used the Lesne model for each different centre as she subdivides hers by centre? Probably best to ask her directly!

**Comment by kaylai** [comment]
Probably! Given time constraints, I'm going to mark this as "future features" so that we can make sure to implement this solution after submission.

**Comment by kaylai** [comment]
@PennyWieser I spoke with Chelsea Allison, who says she used the power equation for Etna from Lesne (2011).

**Comment by kaylai** [comment]
Looked more into this. If we are getting such drastically different results from the Allison spreadsheet, I think there must be something wrong in the code. Penny, were you able to get any results to match up? Or all they all way off?

**Comment by PennyWieser** [comment]
I haven't tried again integrating a H2O model yet- all the testing I did was just using CO2- that was my plan for Friday (sorry, the other testing has taken longer than I imagined and I need to get a paper back to an editor tomorow!).

**Comment by kaylai** [comment]
I tried coding up the Holloway and Blank (1994) model, but it's being very buggy. I'm not sure what the issue is. I'll see if I can fix this this week or next.

**Comment by kaylai** [comment]
I've implemented the Holloway and Blank (1994) fugacity model, and it's working now. I've changed AllisonCarbon to use this model, but the results are still odd. I think we need to rethink how this model is coded or try Penny's suggestion of pairing it with Lesne for H2O.

Commit d3745bb1c97b0b125058ca4e5454ed7963b9f16b

**Comment by PennyWieser** [comment]
Kayla have you tested it with the other H2O models yet or should i do that now? Just making a dataset from her spreadsheet to run through for the different models.

**Comment by kaylai** [comment]
No, I haven’t tested it paired with an H2O model. Worth playing around with.

**Comment by PennyWieser** [comment]
Cool. Will report back
